jerry-core/parser/js/js-parser-expr.c

```diff
@@ -356,8 +357,14 @@ parser_append_object_literal_item (parser_context_t *context_p, /**< context */
     context_p->stack_top_uint8 = PARSER_OBJECT_PROPERTY_BOTH_ACCESSORS;
   }
 } /* parser_append_object_literal_item */
+#endif /* CONFIG_DISABLE_ES2015_OBJECT_INITIALIZER */
 
 #ifndef CONFIG_DISABLE_ES2015_CLASS
+
+#ifdef CONFIG_DISABLE_ES2015_OBJECT_INITIALIZER
+#error "Class support requires ES2015 object literal support"
+#endif /* CONFIG_DISABLE_ES2015_OBJECT_INITIALIZER */
```

Comment:
If we're making this a requirement, should maybe this code be moved to config.h?

Reply:
I searched through the code base and only a few `#error` is there. It seems there is a configuration check in the normal code base:
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/jcontext/jcontext.h#L194

But config.h also have a check.
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/config.h#L81

The truth is I don't know we have a policy for that. It depends config.h is mandatory, or user supplied. Anybody knows this?

Reply:
(In most projects I know config.h is auto generated, and contains no dependency checks. In JerryScript config.h is not auto generated.)